lib/spack/docs/packaging_guide.rst

```diff
@@ -4056,34 +4056,145 @@ Making a package discoverable with ``spack external find``
 
 To make a package discoverable with
 :ref:`spack external find <cmd-spack-external-find>` you must
-define one or more executables associated with the package and must
-implement a method to generate a Spec when given an executable.
+mark it with the ``detectable`` decorator:
+
+.. code-block:: python
+
+   @detectable
+   class Gcc(AutotoolsPackage):
+      """The GNU Compiler Collection includes front ends for C, C++, Objective-C,
+      Fortran, Ada, and Go, as well as libraries for these languages."""
+
+define the executables associated with the package and must
+implement methods to generate one or more Specs when given
+a set of executables.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``executables`` attribute
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 The executables are specified as a package level ``executables``
-attribute which is a list of strings (see example below); each string
-is treated as a regular expression (e.g. 'gcc' would match 'gcc', 'gcc-8.3',
-'my-weird-gcc', etc.).
+attribute which is a list of strings:
+
+.. code-block:: python
+
+   class FooPackage(Package):
+       # Each string provided here is treated as a regular expression, and
+       # would match for example 'foo', 'foobar', and 'bazfoo'.
+       executables = ['foo']
+
+Each string is treated as a regular expression (e.g. 'gcc' would match 'gcc', 'gcc-8.3',
+'my-weird-gcc', etc.) to determine a set of system executables that might be part
+or this package.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Using the default workflow to detect specs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The detection of system installed packages in Spack is based on
```

Comment:
@scheibelp Docs have been reworked in 6e0f32a to follow the structure suggested here: typical use cases first, advanced/rare use cases later (instead of just describing the mechanics of the current implementation).